packages/main/src/Menu.ts

```diff
@@ -606,36 +521,28 @@ class Menu extends UI5Element {
 		}
 	}
 
-	_itemKeyDown(e: KeyboardEvent) {
-		const isMenuClose = this.isRtl ? isRight(e) : isLeft(e);
-		const isMenuOpen = this.isRtl ? isLeft(e) : isRight(e);
+	async _itemKeyDown(e: KeyboardEvent) {
+		const shouldCloseMenu = this.isRtl ? isRight(e) : isLeft(e);
+		const shouldOpenMenu = this.isRtl ? isLeft(e) : isRight(e);
 
 		if (isEnter(e)) {
 			e.preventDefault();
 		}
-		if (isMenuOpen) {
-			const opener = e.target as OpenerStandardListItem;
-			const item = opener.associatedItem;
-			const hoverId = opener.getAttribute("id")!;
-
-			item.hasSubmenu && this._prepareSubMenuDesktopTablet(item, opener, hoverId);
-		} else if (isMenuClose && this._isSubMenu && this._parentMenuItem) {
-			const parentMenuItemParent = this._parentMenuItem.parentElement as Menu;
-			parentMenuItemParent._closeItemSubMenu(this._parentMenuItem, true);
+		if (shouldOpenMenu) {
+			const item = e.target as MenuItem;
+			item.hasSubmenu && await this._prepareSubMenuDesktopTablet(item);
+		} else if (shouldCloseMenu && this._isSubMenu && this._parentMenuItem) {
+			const parentItemMenu = this._parentMenuItem.parentElement as Menu;
```

Comment:
Can we have that interaction trough events? For example the menu to throw some event to the parent menu like "close-submenu" and then the parent to close it. I don't think that reaching to the parent and calling a method of the parent is good practice.